src/linear_algebra/sesquilinear_form.lean

```diff
@@ -133,6 +133,57 @@ lemma map_sum_right {α : Type*} (S : sesq_form R M I) (t : finset α) (g : α 
   S w (∑ i in t, g i) = ∑ i in t, S w (g i) :=
 by haveI s_inst := is_add_monoid_hom_right S w; exact (finset.sum_hom t (λ z, S w z)).symm
 
+variables {M₂ : Type w} [add_comm_group M₂] [module R M₂]
+
+/-- Apply the linear maps `f` and `g` to the left and right arguments of the sesquilinear form. -/
+def comp (S : sesq_form R M I) (f g : M₂ →ₗ[R] M) : sesq_form R M₂ I :=
+{ sesq := λ x y, S (f x) (g y),
+  sesq_add_left := by simp [add_left],
+  sesq_smul_left := by simp [smul_left],
+  sesq_add_right := by simp [add_right],
+  sesq_smul_right := by simp [smul_right] }
+
+/-- Apply the linear map `f` to the left argument of the sesquilinear form. -/
+def comp_left (S : sesq_form R M I) (f : M →ₗ[R] M) : sesq_form R M I :=
+  S.comp f linear_map.id
+
+/-- Apply the linear map `f` to the right argument of the sesquilinear form. -/
+def comp_right (S : sesq_form R M I) (f : M →ₗ[R] M) : sesq_form R M I :=
+  S.comp linear_map.id f
+
+lemma comp_left_comp_right (S : sesq_form R M I) (f g : M →ₗ[R] M) :
+  (S.comp_left f).comp_right g = S.comp f g := rfl
+
+lemma comp_right_comp_left (S : sesq_form R M I) (f g : M →ₗ[R] M) :
+  (S.comp_right g).comp_left f = S.comp f g := rfl
+
+lemma comp_comp {M₃ : Type*} [add_comm_group M₃] [module R M₃]
+  (S : sesq_form R M₃ I) (l r : M →ₗ[R] M₂) (l' r' : M₂ →ₗ[R] M₃) :
+  (S.comp l' r').comp l r = S.comp (l'.comp l) (r'.comp r) := rfl
+
+@[simp] lemma comp_apply (S : sesq_form R M₂ I) (l r : M →ₗ[R] M₂) (v w : M) :
+  S.comp l r v w = S (l v) (r w) := rfl
+
+@[simp] lemma comp_left_apply (S : sesq_form R M I) (f : M →ₗ[R] M) (v w : M) :
+  S.comp_left f v w = S (f v) w := rfl
+
+@[simp] lemma comp_right_apply (S : sesq_form R M I) (f : M →ₗ[R] M) (v w : M) :
+  S.comp_right f v w = S v (f w) := rfl
+
+/-- Let `l`, `r` be surjective linear maps, then two sesquilinear forms are equal if and only if
+  the sesquilinear forms resulted from composing with `l` and `r` are equal. -/
+lemma comp_injective (S₁ S₂ : sesq_form R M₂ I) (l r : M →ₗ[R] M₂)
```

Comment:
```suggestion
lemma comp_injective (S₁ S₂ : sesq_form R M₂ I) {l r : M →ₗ[R] M₂}
```

Reply:
I have also changed the arguments in the corresponding lemma in `linear_algebra.bilinear_form` to implicit also